renderers/ebitengine/ebitengine.go

```diff
@@ -0,0 +1,414 @@
+package ebitengine
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"log/slog"
+	"math"
+	"strings"
+	"unsafe"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/text/v2"
+	"github.com/hajimehoshi/ebiten/v2/vector"
+	"github.com/totallygamerjet/clay"
+)
+
+var (
+	whiteImage *ebiten.Image
+)
+
+func init() {
+	img := ebiten.NewImage(3, 3)
+	img.Fill(color.White)
+	whiteImage = img.SubImage(image.Rect(1, 1, 2, 2)).(*ebiten.Image)
+}
+
+type RendererData struct {
+	Screen *ebiten.Image
+	Fonts  []text.Face
+}
+
+func MeasureText(txt clay.StringSlice, config *clay.TextElementConfig, userData unsafe.Pointer) clay.Dimensions {
+	fonts := *(*[]text.Face)(userData)
+
+	font := fonts[config.FontId]
+
+	width, height := text.Measure(txt.String(), font, font.Metrics().HLineGap)
+	return clay.Dimensions{
+		Width:  float32(width),
+		Height: float32(height),
+	}
+}
+
+func ClayRender(rendererData *RendererData, renderCommands clay.RenderCommandArray) {
+	screen := rendererData.Screen
+	fonts := rendererData.Fonts
+	for i := int32(0); i < renderCommands.Length; i++ {
+		renderCommand := clay.RenderCommandArray_Get(&renderCommands, i)
+		boundingBox := renderCommand.BoundingBox
+		switch renderCommand.CommandType {
+		case clay.RENDER_COMMAND_TYPE_RECTANGLE:
+			config := &renderCommand.RenderData.Rectangle
+			if config.CornerRadius.TopLeft > 0 {
+				RenderFillRoundedRect(screen, boundingBox, config.CornerRadius.TopLeft, config.BackgroundColor)
```

Comment:
let's not ignore the error returned from this function and return it in ClayRender. Also `RenderFillRoundedRect` should be unexported

Reply:
Sure, what to do with the error when we're in `app.Draw`?
Note that these errors/exposition stuff can also be done in sdl2, sdl3 👀